### ISSUE TITLE
Workload 502 Gateway Error

### ISSUE THREAD

**Issue by ckittel** [mspnp/aks-secure-baseline#125]
Hum, I think the workload step doesn't seem to be working like it used to.  I just ran through another end to end, and had to revert to a prior version of `aspnetapp.yaml` for it to work.  It might have simply been something in one of my personal tweaks, but I wanted to leave this issue out here in case others run into this as well.  I do plan on triaging this next week Monday to figure out exactly if it was a fat-finger on my side or is indeed something up with the reference implementation.  So this issue is just advisory for now.  If anyone else is seeing this, feel free to drop a "+1" on here.  If it's work for others, that would be good to know as well.

**Comment by luissalgadofreire** [comment]
Hi, @ckittel. 

I'm seeing it too. 

I took a log of the ingress pods and saw error:

```
{"ingress":"aspnetapp-ingress","level":"error","msg":"Cannot create service: subset not found","namespace":"workloads","providerName":"kubernetes","serviceName":"aspnetapp-service","servicePort":"80","time":"2021-01-02T17:32:32Z"}
```

A quick google and I was directed to [this](https://stackoverflow.com/questions/63713334/cannot-create-service-subset-not-found) entry in stackoverflow regarding traefik. The solution seems to have been adding SecretName to the ingress definition.

I noticed that aspnetapp.yaml has line `# secretName: <bu0001workloads-00-example-com-tls-secret>` commented out.

Could something close to this be the source of the problem?
